pvlib/irradiance.py

```diff
@@ -1212,6 +1225,235 @@ def perez(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
+
+    solar_azimuth : numeric
+        Sun azimuth angles in decimal degrees. solar_azimuth must be >=0
+        and <=360. The azimuth convention is defined as degrees east of
+        north (e.g. North = 0, East = 90, West = 270).
+
+    airmass : numeric (optional, default None)
+        Relative (not pressure-corrected) airmass values. If airmass is a
+        DataFrame it must be of the same size as all other DataFrame
+        inputs. The kastenyoung1989 airmass calculation is used internally
+        and is also recommended when pre-calculating airmass because
+        it was used in the original model development.
+
+    model : string (optional, default='allsitescomposite1990')
+        A string which selects the desired set of Perez coefficients.
+        This argument is only for compatibility with the `perez` function.
+        If a value other than the default value is provided, it is ignored.
+
+    return_components: bool (optional, default=False)
+        Flag used to decide whether to return the calculated diffuse components
+        or not.
+
+    Returns
+    --------
+    numeric, OrderedDict, or DataFrame
+        Return type controlled by `return_components` argument.
+        If ``return_components=False``, `sky_diffuse` is returned.
+        If ``return_components=True``, `diffuse_components` is returned.
+
+    sky_diffuse : numeric
+        The sky diffuse component of the solar radiation on a tilted
+        surface.
+
+    diffuse_components : OrderedDict (array input) or DataFrame (Series input)
+        Keys/columns are:
+            * sky_diffuse: Total sky diffuse
+            * isotropic
+            * circumsolar
+            * horizon
+
+    Notes
+    -----
+    The Driesse-Perez model can be considered a plug-in replacement for the
+    1990 Perez model using the ``'allsitescomposite1990'`` coefficient set.
+    Deviations between the two are very small, as demonstrated in [1]_.
+    Other coefficient sets are not supported because the 1990 set is
+    based on the largest and most diverse set of empirical data.
+
+    References
+    ----------
+    .. [1] A. Driesse, A. Jensen, R. Perez, A Continuous Form of the Perez
+        Diffuse Sky Model for Forward and Reverse Transposition, accepted
+        for publication in the Solar Energy Journal.
+
+    .. [2] Perez, R., Ineichen, P., Seals, R., Michalsky, J., Stewart, R.,
+       1990. Modeling daylight availability and irradiance components from
+       direct and global irradiance. Solar Energy 44 (5), 271-289.
+
+    See also
+    --------
+    perez
+    isotropic
+    haydavies
+    klucher
+    reindl
+    king
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Oct., 2023
+
+    if model not in ('allsitescomposite1990', '1990'):
+        warnings.warn("The 'model' parameter is ignored. "
+                      "Only 'allsitescomposite1990' is available.",
+                      UserWarning)
```

Comment:
I have a mild preference for not having the `model` parameter in the Perez-Driesse model (and state in the docs that it's developed based on `allsitescomposite1990`).

If future model options are developed, it would be easy to add them (but this is not foreseen at the moment).

Reply:
This was motivated by the "plug-in" compatibility claim.  If the upper classes can live without it, so can I.

Reply:
Any more input from those who use and maintain the classes?

Reply:
Regarding the classes: it seems `ModelChain` has no provision for specifying alternative parameter sets, but `PVsystem` allows `model` to be specified via **kwargs by the user.  No big deal to not have `model` for the P-D model in my view.

I suppose I'd lean towards removing `model`, for simplicity and so that those who use alternative parameter sets (which I speculate is rather uncommon) are forced to consider the consequences of switching to the P-D model.

Reply:
Ok, I'll remove it.